spock-spring/src/main/java/org/spockframework/spring/SpringMockTestExecutionListener.java

```diff
@@ -61,6 +64,10 @@ public void beforeTestMethod(TestContext testContext) throws Exception {
     }
   }
 
+  private boolean isSupportedBeanScope(BeanDefinition beanDefinition) {
+    return beanDefinition.isPrototype() || beanDefinition.isSingleton();
```

Comment:
A prototype returns a new bean instance for each `getBean` call, so we would mock a different instance than is injected into other beans. It won't throw an exception, but it would not work as expected either.

Wouldn't this also exclude `@RefreshScope` beans from http://cloud.spring.io/spring-cloud-static/docs/1.0.x/spring-cloud.html#_refresh_scope although it should work?